core/domain/html_cleaner.py

```diff
@@ -300,7 +345,7 @@ def convert_to_text_angular(html_data):
     # Removal of tags can break the soup into parts which are continuous
     # and not wrapped in any tag. This part recombines the continuous
     # parts not wrapped in any tag.
-    soup = bs4.BeautifulSoup(str(soup), 'html.parser')
+    soup = bs4.BeautifulSoup(unicode(soup).encode('utf-8'), 'html.parser')
```

Comment:
I don't think you need the `unicode` cast.

Reply:
``unicode`` cast is needed here because the html strings can contain unicode characters and if we do a ``str`` cast, the ascii codec is unable to decode them.

For example: 
The html
 ``<span><div><span>Hello </span><span></span></div><oppia-noninteractive-collapsible content-with-value=\"&amp;quot;&amp;nbsp;lorem ipsum&amp;quot;\" heading-with-value=\"&amp;quot;lorem ipsum\\&amp;quot;lorem ipsum\\&amp;quot;?&amp;quot;\"></oppia-noninteractive-collapsible><div><span>this is test case for</span></div><div><span>migration </span></div><div><span><br></span></div><div><span>testing </span></div></span>``

is converted to this after migration
``<p>Hello </p><p><oppia-noninteractive-collapsible content-with-value="&amp;quot;&amp;lt;p&amp;gt;\xc2\xa0lorem ipsum&amp;lt;/p&amp;gt;&amp;quot;" heading-with-value="&amp;quot;lorem ipsum\&amp;quot;lorem ipsum\&amp;quot;?&amp;quot;"></oppia-noninteractive-collapsible></p><p>this is test case for</p><p>migration </p><p><br></p><p>testing </p>``

 The ascii codec gives error due to `\xc2\xa0` which is a nobreak space (formed due to conversion of `&nbsp;` written in content-with-value).

Reply:
Ah I see, ok. Make sure to have at least one test with unicode chars and ensure that it looks correct (visually) after the migration.

Reply:
Added the tests and also checked that it looks visually same.